engine/src/main/java/org/terasology/engine/physics/bullet/BulletPhysics.java

```diff
@@ -128,27 +161,227 @@ public BulletPhysics() {
         discreteDynamicsWorld.getBroadphase().getOverlappingPairCache().setInternalGhostPairCallback(ghostPairCallback);
+    }
 
+
+    @ReceiveEvent(components = {RigidBodyComponent.class, LocationComponent.class})
+    public void newRigidBody(OnActivatedComponent event, EntityRef entity, RigidBodyComponent rigidBody, LocationComponent location) {
+        entityRigidBodies.computeIfAbsent(entity, e -> {
+            btCollisionShape shape = getShapeFor(e);
+            if (location != null && rigidBody != null && shape != null) {
+                float scale = location.getWorldScale();
+
+                if (rigidBody.mass < 1) {
+                    logger.warn("RigidBodyComponent.mass is set to less than 1.0, this can lead to strange behaviour, " +
+                            "such as the objects moving through walls. " +
+                            "Entity: {}", e);
+                }
+                Vector3f inertia = new Vector3f();
+                shape.calculateLocalInertia(rigidBody.mass, inertia);
+                shape.setLocalScaling(new Vector3f(scale, scale, scale));
+
+                btRigidBody.btRigidBodyConstructionInfo info =
+                        new btRigidBody.btRigidBodyConstructionInfo(rigidBody.mass,
+                                new EntityMotionState(e),
+                                shape,
+                                inertia);
+                btRigidBody bt = new btRigidBody(info);
+                info.dispose();
+
+                Vector3f temp = new Vector3f();
+                bt.setAngularFactor(temp.set(rigidBody.angularFactor));
+                bt.setLinearFactor(temp.set(rigidBody.linearFactor));
+                bt.setFriction(rigidBody.friction);
+
+                bt.setAngularVelocity(rigidBody.angularVelocity);
+                bt.setLinearVelocity(rigidBody.velocity);
+                bt.userData = e;
+
+                bt.setWorldTransform(
+                        new Matrix4f()
+                                .translationRotateScale(
+                                        location.getWorldPosition(new Vector3f()),
+                                        location.getWorldRotation(new Quaternionf()),
+                                        1.0f));
+
+                discreteDynamicsWorld.addRigidBody(bt,
+                        CollisionGroup.combineGroups(rigidBody.collisionGroup),
+                        CollisionGroup.combineGroups(rigidBody.collidesWith));
+                return bt;
+            } else if (shape != null) {
+                shape.dispose();
+            }
+            return null;
+        });
     }
 
-    public btDiscreteDynamicsWorld getWorld() {
-        return discreteDynamicsWorld;
+
+
+    @ReceiveEvent(components = {RigidBodyComponent.class, LocationComponent.class})
+    public void updateRigidBody(OnChangedComponent event, EntityRef entity, RigidBodyComponent body, LocationComponent location) {
```

Comment:
It's sufficient to list components as parameters.

```suggestion
    @ReceiveEvent
    public void updateRigidBody(OnChangedComponent event, EntityRef entity, RigidBodyComponent body, LocationComponent location) {
```